### DISCUSSION TITLE
How to create the file buffer?

### DISCUSSION THREAD

**Discussion by 4silvertooth** [alfianlensundev/tauri-plugin-printer#8] [Q&A]
What I am doing is as below with proper permissions in conf, but it throws error `File not supported`. [Here](https://github.com/alfianlensundev/tauri-plugin-printer/blob/2cd7243d107589c1a63f2cc91a73c4036a1240d7/dist/index.js#L135)

```js
  import { readBinaryFile, BaseDirectory } from '@tauri-apps/api/fs';
  const buffer = await readBinaryFile("test.pdf", { dir: BaseDirectory.AppData});

  await print_file({
    id: "TWljcm9zb2Z0IFByaW50IHRvIFBERg==",
    path: PATH/TO/APPDATA/, 
    file: buffer,
    print_setting: {
        orientation: "landscape",
        method: "simplex",
        paper: "A4", 
        scale: "noscale",
        repeat: 1
    }
 }) 
```

**Comment by alfianlensundev** [comment]
> What I am doing is as below with proper permissions in conf, but it throws error `File not supported`. [Here](https://github.com/alfianlensundev/tauri-plugin-printer/blob/2cd7243d107589c1a63f2cc91a73c4036a1240d7/dist/index.js#L135)
> 
> ```js
>   import { readBinaryFile, BaseDirectory } from '@tauri-apps/api/fs';
>   const buffer = await readBinaryFile("test.pdf", { dir: BaseDirectory.AppData});
> 
>   await print_file({
>     id: "TWljcm9zb2Z0IFByaW50IHRvIFBERg==",
>     path: PATH/TO/APPDATA/, 
>     file: buffer,
>     print_setting: {
>         orientation: "landscape",
>         method: "simplex",
>         paper: "A4", 
>         scale: "noscale",
>         repeat: 1
>     }
>  }) 
> ```

just change to this
const binaryFile= await readBinaryFile("test.pdf", { dir: BaseDirectory.AppData});
console.log(binaryFile.buffer)
